private/bufpkg/bufmodule/module.go

```diff
@@ -196,32 +196,6 @@ func ModuleToModuleKey(module Module, digestType DigestType) (ModuleKey, error)
 	)
 }
 
-// ModuleToSelfContainedModuleReadBucketWithOnlyProtoFiles converts the Module to a
-// ModuleReadBucket that contains all the .proto files of the Module and its dependencies.
-//
-// Targeting information will remain the same. Note that this means that the result ModuleReadBucket
-// may have no target files! This can occur when path filtering was applied, but the path filters did
-// not match any files in the Module, and none of the Module's files were targeted.
-// It can also happen if the Module nor any of its dependencies were targeted.
-//
-// *** THIS IS PROBABLY NOT THE FUNCTION YOU ARE LOOKING FOR. *** You probably want
-// ModuleSetToModuleReadBucketWithOnlyProtoFiles to convert a ModuleSet/Workspace to a
-// ModuleReadBucket. This function is used for cases where we want to create an Image
-// specifically for one Module, such as when we need to associate LintConfig and BreakingConfig
-// on a per-Module basis for buf lint and buf breaking. See bufctl.Controller, which is likely
-// the only place this should be used outside of testing.
-func ModuleToSelfContainedModuleReadBucketWithOnlyProtoFiles(module Module) (ModuleReadBucket, error) {
-	modules := []Module{module}
-	moduleDeps, err := module.ModuleDeps()
```

Comment:
Resolving the module deps requires walking the full tree. To avoid the performance issue on large repos we no longer resolve all dependencies.